pkg/controller/backup_deletion_controller.go

```diff
@@ -46,16 +46,36 @@ type backupDeletionController struct {
 	deleteBackupRequestLister listers.DeleteBackupRequestLister
 	backupClient              arkv1client.BackupsGetter
 	snapshotService           cloudprovider.SnapshotService
-	backupService             cloudprovider.BackupService
 	bucket                    string
 	restoreLister             listers.RestoreLister
 	restoreClient             arkv1client.RestoresGetter
 	backupTracker             BackupTracker
 	resticMgr                 restic.RepositoryManager
 	podvolumeBackupLister     listers.PodVolumeBackupLister
 
-	processRequestFunc func(*v1.DeleteBackupRequest) error
-	clock              clock.Clock
+	backupStorageDeleter backupStorageDeleter
+	processRequestFunc   func(*v1.DeleteBackupRequest) error
+	clock                clock.Clock
+}
+
+type backupStorageDeleter interface {
```

Comment:
the need to do this kind of thing is an annoying downside to having package-level funcs vs. methods on a struct :/

Reply:
I'll see if I can find something simpler